src_c/imageext.c

```diff
@@ -207,6 +208,21 @@ write_png(const char *file_name, SDL_RWops *rw, png_bytep *rows, int w, int h,
                  PNG_INTERLACE_NONE, PNG_COMPRESSION_TYPE_BASE,
                  PNG_FILTER_TYPE_BASE);
 
+    if (palette) {
+        doing = "set pallete";
+        const int ncolors = palette->ncolors;
+        int i;
+        if (!(color_ptr =
+                  (png_colorp)SDL_malloc(sizeof(png_colorp) * ncolors)))
```

Comment:
Does this need to be freed anywhere?

Reply:
thank you for your comment, I add a commit to free it after set PLTE